qemu/tests/uefishell.py

```diff
@@ -121,21 +123,28 @@ def send_command(self, command, check_result=False, interval=0.5):
 
         :param command: the command string being executed
         :param check_result: the pattern to validate output
+        :param interval: time interval between commands
+        :return if check_result is True, return matched string
         """
         logging.info("Send uefishell command: %s" % command)
         output = self.session.cmd_output_safe(command)
         time.sleep(interval)
-        #Judge if cmd is run successfully via environment variable 'lasterror'
+        # Judge if cmd is run successfully via environment variable 'lasterror'
         last_error = self.params["last_error"]
         env_var = self.session.cmd_output_safe("set")
-        if not re.search(last_error, env_var):
+        if not re.search(last_error, env_var, re.S):
             self.test.fail("Following errors appear %s when running command %s"
                            % (output, command))
         if check_result:
+            value = ""
             for result in check_result.split(", "):
-                if not re.search(result, output):
+                if not re.findall(result, output, re.S):
                     self.test.fail("The command result is: %s, which does not"
-                                   " match the expectation: %s" % (output, result))
+                                   " match the expectation: %s"
+                                   % (output, result))
+                else:
+                    value += re.findall(result, output, re.S)[0]
+            return value
```

Comment:
it's not a good idea to return a value when a condition is met, and no return when a condition is not met, we can return None when check_result is empty or None

Reply:
agree with you. keep the same return types of the function when a condition is met or not. thanks.